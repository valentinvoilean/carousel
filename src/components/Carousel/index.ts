export { CarouselContainer as Carousel } from './container';
