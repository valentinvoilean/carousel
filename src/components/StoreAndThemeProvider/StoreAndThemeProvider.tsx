import React from 'react';
import { Provider } from 'react-redux';
import { ThemeProvider } from 'styled-components';

import configureStore from '../../store';
import { theme } from '../../theme';

export const StoreAndThemeProvider: React.FunctionComponent = ({ children }) => (
  <Provider store={configureStore()}>
    <ThemeProvider theme={theme}>
      <>{children}</>
    </ThemeProvider>
  </Provider>
);
