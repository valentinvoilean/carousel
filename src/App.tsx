import 'normalize.css';
import 'react-placeholder/lib/reactPlaceholder.css';

import React from 'react';
import GoogleFontLoader from 'react-google-font-loader';
import { ThemeProvider } from 'styled-components';

import { Carousel } from './components/Carousel';
import { StyledApp, StyledLink, StyledTitle } from './styled';

import { fonts } from './constants';
import { theme } from './theme';

export const App: React.FunctionComponent = () => {
  const images = [
    'http://lorempixel.com/1024/576/abstract',
    'http://lorempixel.com/1024/576/city',
    'http://lorempixel.com/1024/576/people',
    'http://lorempixel.com/1024/576/transport',
    'http://lorempixel.com/1024/576/animals',
    'http://lorempixel.com/1024/576/food',
    'http://lorempixel.com/1024/576/nature',
    'http://lorempixel.com/1024/576/business',
    'http://lorempixel.com/1024/576/nightlife',
    'http://lorempixel.com/1024/576/sports'
  ];

  return (
    <ThemeProvider theme={theme}>
      <StyledApp>
        <GoogleFontLoader fonts={fonts} />
        <StyledTitle>Infinite Carousel</StyledTitle>
        <Carousel>
          {images.map((url: string, index: number) => (
            <img src={url} key={index} alt={`img-${index}`} />
          ))}
        </Carousel>
        <StyledLink href="http://valentinvoilean.com">by Valentin-Marian Voilean</StyledLink>
      </StyledApp>
    </ThemeProvider>
  );
};
