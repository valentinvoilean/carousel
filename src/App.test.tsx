import React from 'react';
import renderer from 'react-test-renderer';

import { App } from './App';

import { StoreAndThemeProvider } from './components/StoreAndThemeProvider';

describe('App', () => {
  it('renders without crashing', () => {
    const header = renderer
      .create(
        <StoreAndThemeProvider>
          <App />
        </StoreAndThemeProvider>
      )
      .toJSON();
    expect(header).toMatchSnapshot();
  });
});
