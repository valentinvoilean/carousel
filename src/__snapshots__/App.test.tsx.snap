// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without crashing 1`] = `
<div
  className="sc-gzVnrw kHtxYq"
>
  <h1
    className="sc-htoDjs dfiKfh"
  >
    Infinite Carousel
  </h1>
  <div
    className="sc-bdVaJa fwjnIo"
  >
    <div
      className="sc-bwzfXH hAsabR"
      onMouseDown={[Function]}
      onMouseMove={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
    >
      <div
        className="sc-htpNat laMCQG"
        style={
          Object {
            "transform": "translate3d(-8.333333333333334%, 0, 0)",
          }
        }
      >
        <div
          className="sc-bxivhb bFwOSl"
        >
          <img
            alt="img-9"
            src="http://lorempixel.com/1024/576/sports"
          />
        </div>
        <div
          className="sc-bxivhb bFwOSl"
        >
          <img
            alt="img-0"
            src="http://lorempixel.com/1024/576/abstract"
          />
        </div>
        <div
          className="sc-bxivhb bFwOSl"
        >
          <img
            alt="img-1"
            src="http://lorempixel.com/1024/576/city"
          />
        </div>
        <div
          className="sc-bxivhb bFwOSl"
        >
          <img
            alt="img-2"
            src="http://lorempixel.com/1024/576/people"
          />
        </div>
        <div
          className="sc-bxivhb bFwOSl"
        >
          <img
            alt="img-3"
            src="http://lorempixel.com/1024/576/transport"
          />
        </div>
        <div
          className="sc-bxivhb bFwOSl"
        >
          <img
            alt="img-4"
            src="http://lorempixel.com/1024/576/animals"
          />
        </div>
        <div
          className="sc-bxivhb bFwOSl"
        >
          <img
            alt="img-5"
            src="http://lorempixel.com/1024/576/food"
          />
        </div>
        <div
          className="sc-bxivhb bFwOSl"
        >
          <img
            alt="img-6"
            src="http://lorempixel.com/1024/576/nature"
          />
        </div>
        <div
          className="sc-bxivhb bFwOSl"
        >
          <img
            alt="img-7"
            src="http://lorempixel.com/1024/576/business"
          />
        </div>
        <div
          className="sc-bxivhb bFwOSl"
        >
          <img
            alt="img-8"
            src="http://lorempixel.com/1024/576/nightlife"
          />
        </div>
        <div
          className="sc-bxivhb bFwOSl"
        >
          <img
            alt="img-9"
            src="http://lorempixel.com/1024/576/sports"
          />
        </div>
        <div
          className="sc-bxivhb bFwOSl"
        >
          <img
            alt="img-0"
            src="http://lorempixel.com/1024/576/abstract"
          />
        </div>
      </div>
    </div>
    <button
      className="sc-ifAKCX sc-EHOje jBhozE"
      onClick={[Function]}
    />
    <button
      className="sc-ifAKCX dAqXQt"
      onClick={[Function]}
    />
    <div
      className="sc-bZQynM labZrD"
    >
      1
      /
      10
    </div>
  </div>
  <a
    className="sc-dnqmqq jAPEFj"
    href="http://valentinvoilean.com"
  >
    by Valentin-Marian Voilean
  </a>
</div>
`;
